plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -215,6 +217,17 @@ func (plugin *InterfaceConfigurator) ConfigureVPPInterface(iface *intf.Interface
 		}
 	}
 
+	//configure container IP address
+	addr, isIpv6, err := addrs.ParseIPWithPrefix(iface.ContainerIpAddress)
```

Comment:
i think ParseIPWithPrefix returns an error if the input parameter is nil.  It makes container ip address a mandatory parameter and we do not want that. What I suggest:
- create some method called _SetContainerIP(containerIPAddr) error_ outside of ConfigureVPPInterface (the method is already long enough). Move all checks, parsing and vppcalls related to the container IP address to that method. 

- from CreateVPPInterface just call  SetContainerIP which sets the IPif exist, or does nothing. Do not return error if container IP is not set

Reply:
as you wish